probe-rs/src/session.rs

```diff
@@ -432,6 +423,111 @@ impl Session {
         Ok(interface)
     }
 
+    fn reattach_arm_interface(
+        interface: &mut Box<dyn ArmProbeInterface>,
+        debug_sequence: &Arc<dyn ArmDebugSequence>,
+    ) -> Result<(), Error> {
+        use crate::DebugProbe;
+
+        // Temporarily swap out the interface for a Dummy
+        let tmp_interface = Box::new(FakeProbe::default())
+            .try_get_arm_interface()
+            .unwrap();
+        let mut tmp_interface = tmp_interface
+            .initialize(DefaultArmSequence::create())
+            .unwrap();
+
+        std::mem::swap(interface, &mut tmp_interface);
+
+        log::debug!("Re-attaching Probe");
+        let mut probe = tmp_interface.close();
+        probe.detach()?;
+        probe.attach_to_unspecified()?;
+
+        let new_interface = probe.try_into_arm_interface().map_err(|(_, err)| err)?;
+
+        tmp_interface = new_interface.initialize(debug_sequence.clone())?;
+        // swap it back
+        std::mem::swap(interface, &mut tmp_interface);
+
+        log::debug!("Probe re-attached");
+        Ok(())
+    }
+
+    /// Check if the connected device has a custom erase sequence defined
+    pub fn has_custom_erase_all(&self) -> bool {
```

Comment:
I would rather call all the occurences `*_sequence_erase_all` I think. Because I think `custom` can be mistaken very easily. How is the one from the flash algorithm less custom than the one from the sequence?

Reply:
even better!